cpp/src/Slice/Grammar.cpp

```diff
@@ -3470,13 +3469,16 @@ yyparse(void)
                     auto b = dynamic_pointer_cast<Builtin>(constant->type());
                     if (b && b->isIntegralType())
                     {
-                        IceUtil::Int64 v;
-                        if (IceUtilInternal::stringToInt64(constant->value(), v))
+                        try
                         {
+                            std::int64_t v = std::stoll(constant->value(), nullptr, 0);
                             tok = make_shared<IntegerTok>();
                             tok->v = v;
                             tok->literal = constant->value();
                         }
+                        catch (const std::exception&)
+                        {
```

Comment:
Are there any concerns now that we eat the exception?

Reply:
I don't think so, `tok` remains uninitialized and we report the error below.